src/Security/Authentication/Negotiate/src/Internal/LdapAdapter.cs

```diff
@@ -96,9 +97,11 @@ public static async Task RetrieveClaimsAsync(LdapSettings settings, ClaimsIdenti
             }
         }
 
-        private static void GetNestedGroups(LdapConnection connection, ClaimsIdentity principal, string distinguishedName, string groupCN, ILogger logger, IList<string> retrievedClaims)
+        private static void GetNestedGroups(LdapConnection connection, ClaimsIdentity principal, string distinguishedName, string groupCN, ILogger logger, IList<string> retrievedClaims, HashSet<string> processedGroups)
         {
-            var filter = $"(&(objectClass=group)(sAMAccountName={groupCN}))"; // This is using ldap search query language, it is looking on the server for someUser
+            retrievedClaims.Add(groupCN);
```

Comment:
I've moved it here from under `if` clause, but I'm actually not sure.
Basically what I did was compare the results of this code in Linux vs Windows.
I noticed one difference in their behaviour:
1. When using `filter` below on Windows, the results include all groups, including distribution groups (email distribution lists).
2. On Linux, distribution groups are not resolved.

Reply:
Maybe it warrants a separate issue.